### DISCUSSION TITLE
My entire machine loses internet access sometimes after a user disconnects.

### DISCUSSION THREAD

**Discussion by JeYdevoir** [Pryaxis/TShock#2598] [Q&A] [answered]
Gonna be a weird one....

So this doesn't happen all the time, but after the server has been up for about 24 or more hours, someone can join, play for a bit, and disconnect (successfully, they won't notice anything upon leaving) and the moment they leave my entire server machine can no longer connect to the internet until I restart the machine. This happened a while back too, I'll post the exact versions I've experienced this behavior with at the end. The machine is connected to my router via ethernet. 

I've done some light googling on the situation the first time this happened, not sure how to describe the issue properly because I just find mountains of articles of people who can't get people to connect in the first place.

I don't see any particularly useful server logs or windows logs for what matter when I look into the issue. I don't see anything in the router administration window about what happens but I am able to verify the machine's MAC address is decidedly absent from the list of connected devices. What data would be useful to provide to debug the situation? 

Versions of tShock I've experienced this on:
4.5.4
4.5.5
4.5.15
4.5.16

Sucks when this happens because I rely on remoting into the machine to mess around with it, attaching peripherals to use it like a computer is a huge pain. 

Edit: I'm currently running the world on a vanilla server and seeing if that also encounters the issue, but it occurs very inconsistently in the first place so it will take some time before I'll say for certain if it does or does not have the same issue.

**Comment by hakusaro** [comment]
Okay, so, what’s your internet connection type? What’s your router? What’s your operating system version?

**Reply by JeYdevoir** [reply]
1. Fiberoptic connection. (Machine is connected to router via ethernet)
2. Asus RT-AC68U
3. Windows 10 Pro
Version	21H2
Installed on	‎6/‎18/‎2021
OS build	19044.1586
Experience	Windows Feature Experience Pack 120.2212.4170.0

**Reply by hakusaro** [reply]
It’s not CenturyLink or AT&T Fiber right?

**Reply by JeYdevoir** [reply]
No - It's a local ISP called Greenlight Networks. 

Edit: The router itself stays connected as well as all other devices, just the one machine hosting the server loses the ability to connect.

**Reply by hakusaro** [reply]
Are you using an ISP provided router? Is the ONT/OLT provided by a company called “Calyx”?

**Reply by JeYdevoir** [reply]
No. The ISP provided a NOKIA modem, but this is my personal router.

**Reply by hakusaro** [reply]
What about the ONT/OLT?

**Reply by JeYdevoir** [reply]
I had to do some light research to be able to figure out what this question meant exactly and how to answer it, but from what I can tell, "Calyx" is not a name involved with the ISP at all.

**Reply by JeYdevoir** [reply]
....Okay, I think this can be closed. I just had this problem occur on a completely vanilla server after a couple days, meaning this isn't a tShock issue. Apologies for the disruption.

**Comment by tylerjwatson** [comment, marked as answer]
It sounds like to me that you have a bad NIC driver. If specific usage of the network stack is causing the entire machine to go offline until a reboot. 

Terraria is …not the best with packet management and some cheap/bad network drivers can freeze or crash the system when the packet ring buffers are overflowed. I’ve recently had this exact issue where my servers would get knocked offline if the network was placed under sudden large load. Turned out to be an issue with the niu driver.

if you can, update the drivers for the network card in your server box.  Failing that, pick up an intel network card, they are cheap, have good packet management and are rock solid.

**Reply by JeYdevoir** [reply]
I'll look into all of this right after I get out of work! Thank you for such a detailed reply :)

Edit: I want to add, now that I was in that state again, I can confirm the machine can connect to wifi and obtain internet access when this happens, just the Ethernet port becomes a dud. (I've tried disabling/enabling the device, re-seating the ethernet cord, etc. - nothing works until a restart)

**Reply by JeYdevoir** [reply]
Looks like everything is up to date, according to windows. The device I have is `Intel(R) Ethernet Controller (3) I225-V` which is integrated with my motherboard, MSI Z590 PRO WIFI. Unless you're saying integrated cards in general aren't gonna be good and I need a LAN card regardless.

**Reply by tylerjwatson** [reply]
According to https://www.intel.com/content/www/us/en/support/articles/000057261/ethernet-products/gigabit-ethernet-controllers-up-to-2-5gbe.html it seems like there’s a known issue with that chipset and gaming.

Follow the instructions on that page and see if it helps, or if that seems like too much of an effort, I’d pick up a server-grade second hand network card off ebay. They are cheap and _miles_ better than an onboard one.

**Reply by tylerjwatson** [reply]
Pick yourself up an Intel PRO/1000 card off ebay or something. Those things have server grade features like hardware offloading and flow control, and will outlast the apocalypse.

**Reply by JeYdevoir** [reply]
HERE IS THE FULL ANSWER: 

For anyone googling around and finding my issue, this was solved with a new dedicated network card. I figured this issue wasn't worth fixing since I wasn't really torn about not hosting dedicated Terraria, though with the new terraria update I've been wanting to actually host a few servers and found a decent card at a fair price, and switching my ethernet from the integrated port on the motherboard to my dedicated card has alleviated this issue completely.
